rasterio/__init__.py

```diff
@@ -19,7 +19,7 @@
     else:
         if "PATH" in os.environ:
             for p in os.environ["PATH"].split(os.pathsep):
-                if p and glob.glob(os.path.join(p, "gdal*.dll")):
+                if p and (p != ".") and glob.glob(os.path.join(p, "gdal*.dll")):
```

Comment:
Instead of this change, how about changing the next line (line 23) to `os.add_dll_directory(os.path.abspath(p))` ? In some cases `.` might be significant and intended. I believe `add_dll_directory` doesn't object to any particular directory but does require an absolute path.

Please, if you are willing, close this PR and make a new one using `maint-1.3` as the base branch. That way, the fix will get into the upcoming 1.3.5 release.

Reply:
@sgillies I tried both ways when testing out a fix and both ways work. Thanks!